File: java/cyboquatic_workload_20260807/CyboquaticWorkload.java
// File: java/cyboquatic_workload_20260807/CyboquaticWorkload.java
import java.util.Locale;

public final class CyboquaticWorkload {
    private record Frame(String nodeId, double energyReqJ, double deltaVt,
                         double renewableFraction, double recoveredEnergyJ,
                         double waterQualityGain, double knowledgeFactor,
                         double ecoImpactValue, String decision) { }

    private static double clamp01(double value) {
        return Math.max(0.0, Math.min(1.0, value));
    }

    private static Frame evaluate(String nodeId, double energyReqJ, double deltaVt,
                                  double renewableFraction, double recoveredEnergyJ,
                                  double waterQualityGain) {
        double netEnergy = Math.max(0.0, energyReqJ - recoveredEnergyJ);
        double renewable = clamp01(renewableFraction);
        double waterGain = clamp01(waterQualityGain);
        double stability = Math.exp(-Math.max(0.0, deltaVt));
        double energyQuality = Math.exp(-netEnergy / 50000.0);
        double knowledge = clamp01(0.35 * stability + 0.30 * renewable
                + 0.20 * energyQuality + 0.15 * waterGain);
        double ecoImpact = clamp01(0.40 * stability + 0.30 * renewable
                + 0.20 * waterGain + 0.10 * energyQuality);
        boolean accepted = deltaVt <= 0.0 && renewable >= 0.70
                && netEnergy <= 50000.0 && waterGain >= 0.20;
        return new Frame(nodeId, energyReqJ, deltaVt, renewable, recoveredEnergyJ,
                waterGain, knowledge, ecoImpact, accepted ? "ACCEPT" : "REJECT");
    }

    public static void main(String[] args) {
        if (args.length != 6) {
            System.err.println("usage: CyboquaticWorkload NODE ENERGYREQJ DELTAVT "
                    + "RENEWABLE RECOVEREDJ WATERQUALITYGAIN");
            System.exit(1);
        }
        try {
            Frame f = evaluate(args[0], Double.parseDouble(args[1]), Double.parseDouble(args[2]),
                    Double.parseDouble(args[3]), Double.parseDouble(args[4]),
                    Double.parseDouble(args[5]));
            System.out.println("node_id,energyreq_j,delta_vt,renewable_fraction,"
                    + "recovered_energy_j,water_quality_gain,knowledge_factor,"
                    + "eco_impact_value,decision");
            System.out.printf(Locale.ROOT, "%s,%.6f,%.6f,%.6f,%.6f,%.6f,%.6f,%.6f,%s%n",
                    f.nodeId(), f.energyReqJ(), f.deltaVt(), f.renewableFraction(),
                    f.recoveredEnergyJ(), f.waterQualityGain(), f.knowledgeFactor(),
                    f.ecoImpactValue(), f.decision());
            System.exit("ACCEPT".equals(f.decision()) ? 0 : 2);
        } catch (NumberFormatException error) {
            System.err.println("all numeric inputs must be finite decimal values");
            System.exit(1);
        }
    }
}

File: cpp/simulation/cyboquatic_workload_20260807/cyboquatic_workload.cpp
// File: cpp/simulation/cyboquatic_workload_20260807/cyboquatic_workload.cpp
#include <algorithm>
#include <cmath>
#include <cstdlib>
#include <iomanip>
#include <iostream>
#include <sstream>
#include <string>

namespace cyboquatic {

struct Frame {
    std::string node_id;
    double energy_req_j;
    double delta_v_t;
    double renewable_fraction;
    double recovered_energy_j;
    double water_quality_gain;
    double knowledge_factor;
    double eco_impact_value;
    bool accepted;
};

double clamp01(double value) {
    return std::clamp(value, 0.0, 1.0);
}

Frame evaluate(const std::string& node_id, double energy_req_j, double delta_v_t,
               double renewable_fraction, double recovered_energy_j,
               double water_quality_gain) {
    const double net_energy_j = std::max(0.0, energy_req_j - recovered_energy_j);
    const double renewable = clamp01(renewable_fraction);
    const double stability = std::exp(-std::max(0.0, delta_v_t));
    const double energy_quality = std::exp(-net_energy_j / 50000.0);
    const double water_gain = clamp01(water_quality_gain);

    Frame frame{};
    frame.node_id = node_id;
    frame.energy_req_j = energy_req_j;
    frame.delta_v_t = delta_v_t;
    frame.renewable_fraction = renewable;
    frame.recovered_energy_j = recovered_energy_j;
    frame.water_quality_gain = water_gain;
    frame.knowledge_factor = clamp01(
        0.35 * stability + 0.30 * renewable + 0.20 * energy_quality + 0.15 * water_gain);
    frame.eco_impact_value = clamp01(
        0.40 * stability + 0.30 * renewable + 0.20 * water_gain + 0.10 * energy_quality);
    frame.accepted = delta_v_t <= 0.0 && renewable >= 0.70 &&
                     net_energy_j <= 50000.0 && water_gain >= 0.20;
    return frame;
}

std::string csv(const Frame& f) {
    std::ostringstream out;
    out << std::fixed << std::setprecision(6)
        << f.node_id << ',' << f.energy_req_j << ',' << f.delta_v_t << ','
        << f.renewable_fraction << ',' << f.recovered_energy_j << ','
        << f.water_quality_gain << ',' << f.knowledge_factor << ','
        << f.eco_impact_value << ',' << (f.accepted ? "ACCEPT" : "REJECT");
    return out.str();
}

}  // namespace cyboquatic

int main(int argc, char** argv) {
    if (argc != 7) {
        std::cerr << "usage: cyboquatic_workload NODE ENERGYREQJ DELTAVT RENEWABLE "
                     "RECOVEREDJ WATERQUALITYGAIN\n";
        return EXIT_FAILURE;
    }

    try {
        const auto frame = cyboquatic::evaluate(
            argv[1], std::stod(argv[2]), std::stod(argv[3]), std::stod(argv[4]),
            std::stod(argv[5]), std::stod(argv[6]));
        std::cout << "node_id,energyreq_j,delta_vt,renewable_fraction,recovered_energy_j,"
                     "water_quality_gain,knowledge_factor,eco_impact_value,decision\n"
                  << cyboquatic::csv(frame) << '\n';
        return frame.accepted ? EXIT_SUCCESS : 2;
    } catch (const std::exception&) {
        std::cerr << "all numeric inputs must be finite decimal values\n";
        return EXIT_FAILURE;
    }
}
